### PR TITLE
Massive Refactor This Morning

### PR DESCRIPTION
**Pull request by JWURD**

### CODE REVIEW

File: src/main/java/com/zipcodewilmington/phonebook/Person.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    // at some point a better way is using stringbuilder.
+    @Override
+    public String toString() {
+        return "com.zipcodewilmington.phonebook.Person{" +
+                "name='" + name + '\'' +
+                ", number='" + number + '\'' +
+                '}';
+    }
+
```
Comment (xt0fer) [inline src/main/java/com/zipcodewilmington/phonebook/Person.java]: NQR: how does this work for multiple numbers?

File: src/main/java/com/zipcodewilmington/phonebook/PhoneBook.java [hunk 2]
```diff
@@ -1,7 +1,141 @@
...
+     * @param phoneNumber
+     * @return
+     */
+    public Person lookUpPhoneNumber(String phoneNumber) {
+        boolean isPhoneNumber = false;
+        /**reverse lookup need to look value "phoneNumber"*/
+        for (Map.Entry<String, Person> entry : phoneBook.entrySet()) {
+
+            isPhoneNumber = entry.getValue().getListOfNumbers().contains(phoneNumber);
+
```
Comment (xt0fer) [inline src/main/java/com/zipcodewilmington/phonebook/PhoneBook.java]: hereis

File: src/main/java/com/zipcodewilmington/phonebook/PhoneBook.java [hunk 3]
```diff
@@ -1,7 +1,141 @@
...
+    }
+
+    /**
+     * remove a phone number from a list of numbers.
+     */
+    public boolean removeAPhoneNumber(String name, String phoneNumber) {
+        boolean removedANumber = false;
+        try {
+            if (phoneBook.containsKey(name)) {
+              removedANumber =  lookUpPhoneNumber(phoneNumber).getListOfNumbers().remove(phoneNumber);
```
Comment (xt0fer) [inline src/main/java/com/zipcodewilmington/phonebook/PhoneBook.java]: hereis

File: src/main/java/com/zipcodewilmington/phonebook/PhoneBook.java [hunk 4]
```diff
@@ -1,7 +1,141 @@
...
+
+        return personAdded;
+    }
+
+    public boolean addNumberToList(String name, String phoneNumber) {
+        boolean addedNumber = false;
+        /** check for existing phonenumber.
+          there is no phone exisiting number add to list*/
+         if(lookUpPhoneNumber(phoneNumber) == null){
+            addedNumber = lookUpName(name).getListOfNumbers().add(phoneNumber);
```
Comment (xt0fer) [inline src/main/java/com/zipcodewilmington/phonebook/PhoneBook.java]: hereis

**Review by xt0fer — COMMENTED** [review]
for each of "hereis" comments, that is functionality you have the Phonebook doing when it should be a method on Person. you should add a number to a person, remove a number on a person, the PHBK shouldn't be manipulating the arraylist kept inside of Person.

File: src/main/java/com/zipcodewilmington/phonebook/Person.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    public ArrayList<String> getListOfNumbers() {
+        return listOfNumbers;
+    }
+
+    public void setListOfNumbers(ArrayList<String> listOfNumbers){
+        this.listOfNumbers = listOfNumbers;
+    }
+
```
Comment (xt0fer) [inline src/main/java/com/zipcodewilmington/phonebook/Person.java]: nopety, nope. you should not allow another object manipulate things you encap inside of Person.

**Review by kristofer — COMMENTED** [review]
looks good.

**Pull request closed without merging**